packages/cardano-graphql-services/test/TxSubmit/TxSubmitHttpServer.test.ts

```diff
@@ -0,0 +1,76 @@
+import { TxSubmitHttpServer } from '../../src';
+import { TxSubmitProvider } from '@cardano-sdk/core';
+import { getPort } from 'get-port-please';
+import cbor from 'cbor';
+import got from 'got';
+
+const tx = cbor.encode('#####');
+
+describe('TxSubmitHttpServer', () => {
+  let txSubmitProvider: TxSubmitProvider;
+  let txSubmitHttpServer: TxSubmitHttpServer;
+  let port: number;
+  let apiUrlBase: string;
+
+  beforeAll(async () => {
+    port = await getPort();
+    apiUrlBase = `http://localhost:${port}`;
+    txSubmitProvider = { healthCheck: jest.fn(() => Promise.resolve({ ok: true })), submitTx: jest.fn() };
+    txSubmitHttpServer = TxSubmitHttpServer.create({ port }, { txSubmitProvider });
+    await txSubmitHttpServer.initialize();
+    await txSubmitHttpServer.start();
+  });
+
+  afterAll(async () => {
+    await txSubmitHttpServer.shutdown();
+  });
+
+  afterEach(async () => {
+    jest.resetAllMocks();
+  });
+
+  it('health', async () => {
+    const res = await got(`${apiUrlBase}/health`, {
+      // eslint-disable-next-line sonarjs/no-duplicate-string
+      headers: { 'Content-Type': 'application/json' }
+    });
+    expect(res.statusCode).toBe(200);
+    expect(JSON.parse(res.body)).toEqual({ ok: true });
+  });
+
+  describe('submit', () => {
+    it('returns a 200 coded response with a well formed HTTP request', async () => {
+      const res = await got.post(`${apiUrlBase}/submit`, {
+        body: tx,
+        headers: { 'Content-Type': 'application/cbor' },
+        method: 'post'
+      });
+      expect(res.statusCode).toBe(200);
+      expect(txSubmitProvider.submitTx).toHaveBeenCalledTimes(1);
+    });
+
+    it('returns a 400 coded response if the wrong content type header is used', async () => {
+      try {
+        await got.post(`${apiUrlBase}/submit`, {
+          body: tx,
+          headers: { 'Content-Type': 'application/json' }
+        });
+      } catch (error) {
+        expect(error.message).toBe('Response code 400 (Bad Request)');
+        expect(txSubmitProvider.submitTx).toHaveBeenCalledTimes(0);
+      }
+    });
+
+    it('returns a 400 coded response if the tx is not sent as binary data', async () => {
+      try {
+        await got.post(`${apiUrlBase}/submit`, {
+          body: Buffer.from(new Uint8Array()).toString(),
+          headers: { 'Content-Type': 'application/json' }
+        });
+      } catch (error) {
```

Comment:
Tests asserting in catch will pass if it doesn't throw. Should either `throw new Error('fail')` after the operation in try block, or do `expect(() => got.post(...)).toThrow('Response code 400 (Bad Request)')`